utils/post_utils.jsx

```diff
@@ -309,6 +313,28 @@ export function getLatestPostId(postIds) {
 export function createAriaLabelForPost(post, author, isFlagged, reactions, intl) {
     const {formatMessage, formatTime, formatDate} = intl;
 
+    const emojiMap = getEmojiMap(store.getState());
+    let message = post.message;
+    let match;
+
+    // Match all the shorthand forms of emojis first
+    for (const name of Object.keys(Emoticons.emoticonPatterns)) {
+        const pattern = Emoticons.emoticonPatterns[name];
+        message = message.replace(pattern, `:${name}:`);
+    }
+
+    while ((match = Emoticons.EMOJI_PATTERN.exec(message)) !== null) {
```

Comment:
@hmhealey thoughts about this? to me it seems a bit overkill as I think we are running something similar to this somewhere else

Reply:
It does feel a bit like overkill since we're parsing emoticons here when we're also parsing for emoticons and markdown in general elsewhere, but I can't think of a better way to do this because those serve different purposes and live in different places. Ideally, we wouldn't repeat this logic, but that would require a large amount of restructuring to solve.

The only issue I can see with doing it this way is that it doesn't account for Markdown in the message text. Again though, I can't think of a way to deal with that without some substantial restructuring, and I don't know how bad that would end up being in the long run

Reply:
yeah my worry is that this will definitely affect client side performance, but I can't think of a better way to do it..

Reply:
It would be nice if there was some way to generate the aria-label for a post only when needed, but I assume there's no way to detect a screen reader or something similar

Reply:
no other way to do this? I still think this is overkill